### ISSUE TITLE
Trying to understand walk-legs within taxi and drtdoor2door trips

### ISSUE THREAD

**Issue by p3nt0xide** [matsim-org/matsim-maas#23]
Hi, 

first off, thank you for sharing your work on this maas extension.

While running your demo scenarios (taxi/drtdoor2door/drtstopbased) I found that both taxi and drtdoor2door trips have walk legs which don't really makes sense to me since start and end link are exactly the same and waiting time for the vehicles is included in the taxi/drt leg.

While I do understand that agents still have to get outside their home to get to the street to be able to be picked-up and have to get to their facility once dropped of by the taxi, these walking times seem oddly large (see screenshot of sample agent)

![Screenshot 2021-11-17 115718](https://user-images.githubusercontent.com/57716371/142188043-26641964-07d4-4061-8228-6ac4a070d537.png)

Is there any explanation for this?

**Comment by kainagel** [comment]
I *think* that this is just the way that it works internally ... accessEgressRouting ... we assume that agents have to walk from facilities to links and vice versa.  The walk distance (and thus the walk time) should be computed from the facility coordinate to some imaginary point on the link.  (The reason why this should be somewhat realistic is that we do not want people to move facilities to totally different coordinates, leave the "link attachment" the same, and then allow agents to move from facility to link in very short time --- thus producing "black holes" in the simulation.)

This may have weird consequences because the small-scale data model here is not completely clean.  We normally say that MATSim does not have good spatial resolution at scales smaller than links ... so if you need more spatial resolution, make your links shorter (possibly artificially in some area of interest).  

We agree that this could be done better, and there is a first mini-step in one of the events that contains an "offsetFromNode" field.  But it ain't easy, since every change here breaks lots of regression tests.

**Comment by p3nt0xide** [comment]
Hi, 

thank you for your answer, this clears the picture a bit for me. Let me allow two follow-up questions: 

1) In the OpenBerlin scenario, I find that accessEgressRouting is enabled for every mode except for bicycle/walk. Other openly accessible scenarios such as the digital twin for Santiago de Chile seem to only have it for public transport (and here Im not sure whether the walking legs are just to get from the transit station to the link and not some random coordinate on it). When I implement the DRT extension in the latter scenarios, however, accessEgressRouting for the taxi mode is enabled which obviously distorts the calibration process. 

Long story short: Is there a possibility to turn off/on accessEgressRouting for certain modes? 

Best regards and thank you!

**Comment by tschlenther** [comment]
Hi, no you can not enable/disable that for specific modes. The Santiago scenario was established before accessEgressRouting was set as a standard. However, public transport (for transfers, but also for access and egress) as well as DRT had "auto-inserted" walk legs independently from the accessEgressRouting setting, even before the latter existed. I suggest to concentrate on the trip main mode instead of the leg mode for calibration, but this might depend on the available data.

**Comment by p3nt0xide** [comment]
Thank you for your reply and your insights on that matter. So speaking shortly, is accessEgressRouting a matter of the matsim version? 

Right, focusing on the main mode travel times (taxi) was my initial thought when calibrating the model. However, this does not really help when running the scenario with modechanging strategies such as subtourModeChoice since the additional walking times make the taxi mode relatively more unattractive in comparison to other modes. 

Do you have any thoughts on how to cope with this? A first idea of mine would be to scale up the taxi utility parameters so that it would offset the additional walking ...
